scripts/build.py

```diff
@@ -598,17 +607,78 @@ def build_files(source, target, file_hashes):
                 continue
             target_path = source_path.replace(source, target)
             ensure_directory_exists(target_path)
-
-            task = threading.Thread(
-                target=minify_func,
-                args=(source_path, target_path, file_hashes, filename))
+            task = threading.Thread()
+            if file_formats:
+                if any(filename.endswith(p) for p in file_formats):
+                    task = threading.Thread(
+                        target=minify_func,
+                        args=(source_path, target_path, file_hashes, filename))
+            else:
+                task = threading.Thread(
+                    target=minify_func,
+                    args=(source_path, target_path, file_hashes, filename))
             tasks.append(task)
         try:
             _execute_tasks(tasks)
         except Exception as e:
             print e
 
 
+def rebuild_new_files(
+        source_path, target_path, recently_changed_filenames, file_hashes):
+    """Minify recently changed files.
+
+    Args:
+        source_path: str. Path relative to /oppia directory of directory
+            containing files and directories to be copied.
+        target_path: str. Path relative to /oppia directory of directory where
+            to copy the files and directories.
+        recently_changed_filenames: list(str). List of filenames that were
+            recently changed.
+        file_hashes: dict(str, str). Dictionary of file hashes.
+    """
+    for file_name in recently_changed_filenames:
```

Comment:
Are we planning on multithreading this? Or is there just that many files that are rebuilt each time?

Reply:
Yep. I have some work in the pipeline that multi-thread almost everything, except the hashing and build_third_party_libs(). If you are just dying to see my latest un-reviewed works, you can check it out [here](https://github.com/oppia/oppia/compare/develop...hoangviet1993:viet_branch_TravisMultithread).

It is also ran on my own Travis [here](https://travis-ci.org/hoangviet1993/oppia/builds/414725515)

Reply:
I would leave those multi-thread changes to a later PR though. Multi-threading won't help out on Travis. Locally, it cuts out 30-50 seconds comparing to this PR.